cpp/doxygen/developer_guide/TESTING.md

```diff
@@ -91,7 +91,7 @@ TYPED_TEST(TypedTestFixture, FirstTest){
 }
 ```
 
-To specify the list of types to use, instead of GTest's `::testing::Types<...>`, libcudf provides `cudf::test::Types<...>` which is a custom, drop-in replacement for `::testing::Types`.
+To specify the list of types to use, instead of GTest's `testing::Types<...>`, libcudf provides `cudf::test::Types<...>` which is a custom, drop-in replacement for `testing::Types`.
```

Comment:
I'm a little confused about why the changes to the `::testing` namespace were needed. The code appears to mix styles between `::testing` and `testing`.

Reply:
It's normal in gtest code to use `::testing::...`  This is the style used in the gtest documentation. I'm not sure why, though...

Reply:
My guess is it's a matter of being explicit and avoiding namespace clashes since it's probably quite common for other projects making use of gtest to have their own `testing` namespaces. The leading `::` guarantees global scope resolution.

As for why I had to change it, I am not entirely sure what doxygen's scope resolution rules are but in some cases (like this one) the leading `::` leads to doxygen failing to find the right link. I get errors like this from the pre-commit doxygen check:
```
cudf/cpp/doxygen/developer_guide/TESTING.md:67: warning: explicit link request to 'testing::Test' could not be resolved
```
I chose to remove the leading colons for expediency to get past this issue since I really have no idea how to resolve it more generally. My best guess is that the global namespace in the context of this document is some doxygen namespace that doesn't contain anything, so the name lookup fails. I don't know how else to resolve this, and I think it's a minor enough issue that it's OK to move forward anyway. If you can find a better solution I'm all for it, though.

Reply:
I don't want to hold up the PR on this but let's leave this conversation unresolved.